src/PwDynamicDefinition.jl

```diff
@@ -26,18 +22,14 @@ This is a tricky case that must be dealt with.
 
 Enclosures `Y[1], Y[2]` for f(a), f(b) and `increasing` may be provided (for instance if we know that `Y=(0,1)`), otherwise they are computed automatically.
 """
-struct MonotonicBranch{T<:Function, U<:Function, S<:Interval}
+struct MonotonicBranch{T<:Function, S<:Interval}
     f::T
-    fprime::U
```

Comment:
I like it here also: taking away the derivative from the MonotonicBranch struct makes it easier to spot errors; in my specific case, many of my code errors are of PEBCAK type, and this guarantees a good default solution.
Thank you @fph

Reply:
A next step to avoid user error could be testing numerically that the provided derivatives make sense: inside the macro `@define_with_derivatives`, one  chooses at random $x$ and $h\approx 10^{-5}$, and checks that $f(x+h) - f(x) - f'(x)h - f''(x)h^2/2$ is of the order of $10^{-15}$. What do you think about it?

Reply:
I have added a function `check_derivatives`. Unfortunately we cannot have it run automatically after `@define_with_derivatives` because the code does not know a valid `x` in the domain of `f` to check the derivative in; the function could be defined only in a very small interval. But it is there, and the docs recommend running it.